compiler/rustc_metadata/src/dependency_format.rs

```diff
@@ -210,19 +211,25 @@ fn calculate_type(tcx: TyCtxt<'_>, ty: CrateType) -> DependencyList {
 
     // Collect what we've got so far in the return vector.
     let last_crate = tcx.crates(()).len();
-    let mut ret = (1..last_crate + 1)
-        .map(|cnum| match formats.get(&CrateNum::new(cnum)) {
-            Some(&RequireDynamic) => Linkage::Dynamic,
-            Some(&RequireStatic) => Linkage::IncludedFromDylib,
-            None => Linkage::NotLinked,
-        })
-        .collect::<Vec<_>>();
+    let mut ret = IndexVec::new();
+    assert_eq!(ret.push(Linkage::Static), LOCAL_CRATE);
```

Comment:
Setting `Linkage::Static` for `LOCAL_CRATE` allows removing a special case in `for_each_exported_symbols_include_dep` and technically the local crate is always statically linked into the linker output even when building a dylib.

Reply:
Suggestion: can you include your review comment inline as a code comment? This is not immediately obvious, and can be tricky to reason about.